crates/ty_ide/src/symbols.rs

```diff
@@ -401,27 +416,94 @@ struct SymbolTree {
     kind: SymbolKind,
     name_range: TextRange,
     full_range: TextRange,
+    import_kind: Option<ImportKind>,
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, get_size2::GetSize)]
+enum ImportKind {
+    Normal,
+    RedundantAlias,
+    Wildcard,
 }
 
 /// A visitor over all symbols in a single file.
 ///
 /// This guarantees that child symbols have a symbol ID greater
 /// than all of its parents.
-struct SymbolVisitor {
+#[allow(clippy::struct_excessive_bools)]
+struct SymbolVisitor<'db> {
+    db: &'db dyn Db,
+    file: File,
     symbols: IndexVec<SymbolId, SymbolTree>,
     symbol_stack: Vec<SymbolId>,
-    /// Track if we're currently inside a function (to exclude local variables)
+    /// Track if we're currently inside a function.
     in_function: bool,
+    /// Track if we're currently inside a class.
+    in_class: bool,
     global_only: bool,
+    /// The origin of an `__all__` variable, if found.
+    all_origin: Option<DunderAllOrigin>,
+    /// A set of names extracted from `__all__`.
+    all_names: FxHashSet<String>,
```

Comment:
nit: most entries in `__all__` are likely to be small -- could we use `ruff_python_ast::Name` here, for less memory allocation?